core/src/main/java/com/netflix/conductor/core/execution/ParametersUtils.java

```diff
@@ -161,23 +161,23 @@ public Object replace(String paramString) {
 
     @SuppressWarnings("unchecked")
     private Map<String, Object> replace(Map<String, Object> input, DocumentContext documentContext, String taskId) {
+        Map<String, Object> result = new HashMap<>();
         for (Entry<String, Object> e : input.entrySet()) {
+            Object newValue;
             Object value = e.getValue();
             if (value instanceof String) {
-                Object replaced = replaceVariables(value.toString(), documentContext, taskId);
-                e.setValue(replaced);
+                newValue = replaceVariables(value.toString(), documentContext, taskId);
             } else if (value instanceof Map) {
```

Comment:
This block of code is where the previous fix falls apart, the new test I added fails without the new fix.